src/partr.c

```diff
@@ -0,0 +1,340 @@
+// This file is a part of Julia. License is MIT: https://julialang.org/license
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <strings.h>
+
+#include "julia.h"
+#include "julia_internal.h"
+#include "gc.h"
+#include "threading.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define JULIA_ENABLE_PARTR
+
+#ifdef JULIA_ENABLE_THREADING
+
+// GC functions used
+extern int jl_gc_mark_queue_obj_explicit(jl_gc_mark_cache_t *gc_cache,
+                                         jl_gc_mark_sp_t *sp, jl_value_t *obj);
+
+// thread sleep threshold
+extern uint64_t jl_thread_sleep_threshold;
+
+// multiq
+// ---
+
+/* a task heap */
+typedef struct taskheap_tag {
+    jl_mutex_t lock;
+    jl_task_t **tasks;
+    int16_t ntasks, prio;
+} taskheap_t;
+
+/* multiqueue parameters */
+static const int16_t heap_d = 8;
+static const int heap_c = 4;
```

Comment:
Is this the same as `MULTIQ_HEAP_C `?

Reply:
It is. Should I move to using `#define`s like `MULTIQ_HEAP_C` or leave this `const`?